### PR TITLE
fix(deps): swap octocrab jwt-rust-crypto for jwt-aws-lc-rs; remove RUSTSEC-2023-0071 advisory

### PR DESCRIPTION
**Pull request by clouatre**
## Summary

Swap the octocrab `jwt-rust-crypto` feature for `jwt-aws-lc-rs` and remove the `RUSTSEC-2023-0071` advisory ignore from `deny.toml`.

octocrab 0.54.1 requires exactly one JWT crypto backend at compile time (`compile_error!` enforces this). The `jwt-rust-crypto` backend pulls in `rsa 0.9.x` which carries [RUSTSEC-2023-0071](https://rustsec.org/advisories/RUSTSEC-2023-0071) (Marvin Attack timing side-channel). Swapping to `jwt-aws-lc-rs` uses `aws-lc-rs` instead, dropping `rsa` entirely from the dependency tree.

No GitHub App authentication code exists in the codebase; only `Octocrab::builder().personal_token()` is used. The swap is a crypto-backend substitution with no runtime behaviour change.

Also removes the `tokio-test` workspace dev-dependency which has zero callers across the entire workspace.

## Changes

- `Cargo.toml`: `jwt-rust-crypto` → `jwt-aws-lc-rs` in octocrab features; remove `tokio-test` from workspace dev-dependencies
- `deny.toml`: remove RUSTSEC-2023-0071 ignore block and comment
- `Cargo.lock`: regenerated; `rsa`, `ed25519-dalek`, `p256`, `p384`, and the sha2-0.10 crypto cluster drop from the tree

## Verification

```
cargo deny check advisories  # advisories ok (zero ignores needed)
cargo deny check bans         # bans ok (sha2/block-buffer/crypto-common/digest/cpufeatures duplicates dropped)
cargo build --workspace       # clean
cargo nextest run --workspace # 590/590 passed, 1 skipped
cargo clippy --profile ci -- -D warnings -W clippy::cognitive_complexity  # clean
```

## Audit reference

Findings C-4, C-6 from `docs/audit/2026-08-06-rust-1.97-idioms-cargo-ci.md`

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -22,7 +22,7 @@ aptu-core = { path = "crates/aptu-core", version = "0.10.0" }
 async-trait = "0.1"
```
[the remark this answers is not in the document]
Comment (clouatre) [reply Cargo.toml:22]: Acknowledged. Cargo.lock pins the resolved version; patch updates only land on an explicit `cargo update` followed by a passing CI run. No action needed.
